_sources/functions_in_cpp/mixed_up_code.rst

```diff
@@ -0,0 +1,344 @@
+   =====
+      double pi = 3.14;
+   =====
+      return pi * radius * radius * height;
+   =====
+   }
+
+.. parsonsprob:: functions_in_cpp_mixed_up_code_6
+   :numbered: left
+   :adaptive:
+   :practice: T
+
+   Let's write a function called ``scale`` which takes an input ``x`` and scales it by a factor 
+   of 2 is ``x`` is even and a factor of 3 if ``x`` is odd. The function should modify the original
+   ``x`` variable, not return the scaled value.
+   -----
+   void scale(int &x) {
+   =====
+   void scale(int x) {  #distractor
+   =====
+   int scale(int x) {  #distractor
+   =====
+      if (x % 2 == 0) {
+   =====
+      if (x % 2) {  #paired
+   =====
+         x = x * 2;
+   =====
+      }
+   =====
+      return x * 2;
+   =====
+      else {
+   =====
+         x = x * 3;
+   =====
+         return x * 3;  #distractor
+   =====
+      }
+   =====
+   }
+
+.. parsonsprob:: functions_in_cpp_mixed_up_code_7
+   :numbered: left
+   :adaptive:
+   :practice: T
+
+   Let's write the code for the isDoubleDigit function. isDoubleDigit
+   takes num as a parameter. isDoubleDigit returns true if 
+   num is a double digit number and returns false otherwise. 
+   In the code, use the if then else structure. 
+   Put the necessary blocks of code in the correct order.
+   -----
+   bool isDoubleDigit (int num) {
+   =====
+   isDoubleDigit (int num) {  #paired
+   =====
+      if (num >= 10 && num < 100) {
+   =====
+      if (10 <= num <= 99) {  #distractor
+   =====
+      if (num > 10 && num < 100) {  #distractor
+   =====
+      if (num > 10 && num <= 100) {  #distractor
+   =====
+         return true;
+   =====
+      }
+   =====
+      else {
+   =====
+         return false;
+   =====
+      }
+   =====
+   }
+
+.. parsonsprob:: functions_in_cpp_mixed_up_code_8
+   :numbered: left
+   :adaptive:
+   :practice: T
+
+   Let's write the code for the Compare function. Compare
+   takes two integers a, b. Compare returns 1 if 
+   a is greater than b, -1 if a is less than b and 0 if they are equal.
+   Put the necessary blocks of code in the correct order.
+   -----
+   int Compare (int a, int b) {
+   =====
+   bool Compare (int a, int b) {   #paired
+   =====
+      if (a > b) {
+   =====
+      if (a > b && a &lt b) {  #distractor
+   =====
+         return 1;
+   =====
+      }
+   =====
+      else if (a &lt b) {  
+   =====
+         return -1;
+   =====
+      }
+   =====
+      else if (a > 0){ #distractor
+   =====
+      else {
+   =====
+         return 0;
+   =====
+      }
+   =====
+   }
+
+.. parsonsprob:: functions_in_cpp_mixed_up_code_9
+   :numbered: left
+   :adaptive:
+   :practice: T
+
+   Let's write the code for the isFactor function. isFactor
+   takes two parameters, num and factor.
+   isFactor returns true if factor is a factor of num 
+   and returns false otherwise. In the code, use the if then 
+   else structure. Put the necessary blocks of code in the 
+   correct order.
```

Comment:
I think it would be good to include a brief description about how we know if one number is a factor of another number (in case people have forgotten from math class)